pkg/daemon/manager/manager.go

```diff
@@ -150,6 +151,37 @@ func (m *manager) EnsureServiceState(service winsvc.Service, state svc.State) er
 	return winsvc.WaitForState(service, state)
 }
 
+// Stop the service, and wait for the process associated with the service to stop
+func (m *manager) stopServiceAndProcess(winSvc *mgr.Service) error {
+	status, err := winSvc.Query()
+	if err != nil {
+		return errors.Wrap(err, "error querying service")
+	}
+	var pHandle windows.Handle
+	// A value of 0 indicates that no process is running
+	if status.ProcessId != 0 {
+		pHandle, err = windows.OpenProcess(windows.PROCESS_TERMINATE|windows.PROCESS_QUERY_INFORMATION, false,
+			status.ProcessId)
+		if err != nil {
+			return errors.Wrap(err, "unable to open service's associated process")
+		}
+		defer windows.CloseHandle(pHandle)
+	}
```

Comment:
Could the above suggestion help merge this block with the other `if status.ProcessId != 0 {` below?